### DISCUSSION TITLE
Trying to create catalog group via UrlReference.

### DISCUSSION THREAD

**Discussion by angelikaSajani** [TerriaJS/terriajs#7053] [Q&A] [answered]
Forgive me if this is a stupid question, but I am fairly new to TerriaMap. In my adaptation I need to load a catalog from our server which depends on who is logged in. The correct class to use seemed to me UrlReference, so in my simple.json file I added the following member:

`    {
      "id": "Arbormeta Data",
      "type": "url-reference",
      "name": "Arbormeta Datasets",
      "url": "http://127.0.0.1:8043/api/v1/terria/catalog.json",
      "allowLoad": true,
      "description": "Temporary Description",
      "dataCustodian": "ArborMeta",
      "isGroup": true,
      "isFunction": false,
      "isMappable": false,
      "isChartable": false
    },
` 
The response to the above url is (greatly simplified for testing):

`{
    "catalog": [
        {
            "id": "Arbormeta Data",
            "type": "group",
            "name": "Arbormeta Datasets",
            "sortMembersBy": "id",
            "isOpenInWorkbench": true,
            "isOpen": true,
            "members": [
                {
                    "id": "Brindingabba",
                    "type": "group",
                    "name": "Brindingabba",
                    "sortMembersBy": "id",
                    "isOpenInWorkbench": true,
                    "members": [
                        {
                            "type": "kml",
                            "name": "Brindingabba Boundary",
                            "url": "tjs/plot/Brindingabba/Brindigabba.kml",
                            "isOpenInWorkbench": false
                        }
                    ]
                }
            ]
        }
    ]
}`

The item is recognised in the catalog, but when I am trying to expand it, I get an error:
`at GeosonCatalogItem._callee$ (webpack-internal:///node modules/terriajs/Lib/ModelMixins/MappableMixin.ts:193:69)` telling me the item contains invalid GeoJSON. Full error stack in attached attached file.
![Screenshot 2024-02-19 at 4 33 06 pm](https://github.com/TerriaJS/terriajs/assets/71240254/8c4c57f9-c003-4d6f-b59e-11f0108c6f4c)

Well, it's not supposed to contain any GeoJSON. From looking at the documentation, one would assume that setting 'isMappable' to false would prevent UrlReference from trying to interpret the result as GeoJSON. 

Is there anything obvious that I am doing wrong? Am I supposed to implement my own subclass of UrlReference, or something similar? 

For what it's worth, if I put the json returned by the above url into a text file and manually add it as local data it works fine. When I manually add the link as webdata (type TerriaCatalog) it also works fine (well, mostly, it seems not to be able to determine the optimal zoom), but it displays the one kml file correctly. So the problem must be with UrlReference.

**Comment by angelikaSajani** [comment]
PS: Adding the URL to 'initializationsUrls' hangs the server - the app seems to start up then the screen turns white and hangs. And to preempt any suggestions - CORS issues are not the cause.

**Comment by angelikaSajani** [comment, marked as answer]
I solved the problem, hopefully someone else can benefit from this: I had to use TerriaReference (type 'terria-reference' instead of UrlReference.

**Reply by nf-s** [reply]
Hi @angelikaSajani 

Yes TerriaReference is the correct type to use here. We are in the process of improving our documentation to include examples of the different catalog item types to hopefully make it easier to find the correct types.

Thanks for creating the post!
